akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+trait Dialect {
+  val createTableStatement: String
+  val alterTableStatement: String
+
+  val insertOrUpdateStatement: String
+  val readOffsetQuery: String
+  val clearOffsetStatement: String
+
+}
+
+/* Defines the basic statements. Dialects may use it or define their own.
+ */
+object DialectDefaults {
+  def readOffsetQuery(table: String) =
+    s"SELECT * FROM $table WHERE PROJECTION_NAME = ?"
+
+  def clearOffsetStatement(table: String) =
+    s"DELETE FROM $table WHERE projection_name = ? AND projection_key = ?"
```

Comment:
```suggestion
    s"DELETE FROM $table WHERE PROJECTION_NAME = ? AND PROJECTION_KEY = ?"
```

[nit]

Reply:
Shouldn't we have a `DeleteIndices` or some equivalent solution to populate all prepared statements?